modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/RebalanceRaftGroupEventsListener.java

```diff
@@ -252,92 +294,189 @@ private void scheduleChangePeers(List<PeerId> peers, long term) {
 
-        List<ClusterNode> resolvedNodes = new ArrayList<>(peers.size());
+            // eq(revision(assignments.switch.reduce), retrievedAssignmentsSwitchReduce.revision)
+            SimpleCondition con3 = switchReduceEntry.empty()
+                    ? notExists(switchReduceKey) : revision(switchReduceKey).eq(switchReduceEntry.revision());
 
-        for (PeerId p : peers) {
-            var addr = NetworkAddress.from(p.getEndpoint().getIp() + ":" + p.getEndpoint().getPort());
+            // eq(revision(assignments.switch.append), retrievedAssignmentsSwitchAppend.revision)
+            SimpleCondition con4 = switchAppendEntry.empty()
+                    ? notExists(switchAppendKey) : revision(switchAppendKey).eq(switchAppendEntry.revision());
 
-            if (resolveRegistry.containsKey(addr)) {
-                resolvedNodes.add(resolveRegistry.get(addr));
+            Condition retryPreconditions = and(con1, and(con2, and(con3, con4)));
+
+            tblConfiguration.change(ch -> {
+                List<List<ClusterNode>> assignments = ByteUtils.fromBytes(((ExtendedTableChange) ch).assignments());
+                assignments.set(partNum, stable);
+                ((ExtendedTableChange) ch).changeAssignments(ByteUtils.toBytes(assignments));
+            }).get(10, TimeUnit.SECONDS);
+
+            Update successCase;
+            Update failCase;
+
+            byte[] stableByteArray = ByteUtils.toBytes(stable);
+            byte[] additionByteArray = ByteUtils.toBytes(calculatedPendingAddition);
+            byte[] reductionByteArray = ByteUtils.toBytes(calculatedPendingReduction);
+            byte[] switchReduceByteArray = ByteUtils.toBytes(calculatedSwitchReduce);
+            byte[] switchAppendByteArray = ByteUtils.toBytes(calculatedSwitchAppend);
+
+            if (!calculatedSwitchAppend.isEmpty()) {
+                successCase = ops(
+                        put(stablePartAssignmentsKey, stableByteArray),
+                        put(pendingPartAssignmentsKey, additionByteArray),
+                        put(switchReduceKey, switchReduceByteArray),
+                        put(switchAppendKey, switchAppendByteArray)
+                ).yield(SWITCH_APPEND_SUCCESS);
+                failCase = ops().yield(SWITCH_APPEND_FAIL);
+            } else if (!calculatedSwitchReduce.isEmpty()) {
+                successCase = ops(
+                        put(stablePartAssignmentsKey, stableByteArray),
+                        put(pendingPartAssignmentsKey, reductionByteArray),
+                        put(switchReduceKey, switchReduceByteArray),
+                        put(switchAppendKey, switchAppendByteArray)
+                ).yield(SWITCH_REDUCE_SUCCESS);
+                failCase = ops().yield(SWITCH_REDUCE_FAIL);
             } else {
-                throw new IgniteInternalException("Can't find appropriate cluster node for raft group peer: " + p);
+                Condition con5;
```

Comment:
Variable names look odd

Reply:
well they're just parts of one big IF statement's condition basically, I don't think that they deserve a proper naming. Each of them have a comment that describes the condition